File: solutions/beecrowd/1165/1165.cpp
#include <cmath>
#include <cstdint>
#include <iostream>

bool is_prime(std::int32_t n) {
    if (n == 2) {
        return true;
    }
    if (n % 2 == 0 || n < 2) {
        return false;
    }

    for (std::int32_t i = 3; i <= sqrt(n); i += 2) {
        if (n % i == 0) {
            return false;
        }
    }
    return true;
}

int main() {
    std::int32_t x;
    std::int16_t n;

    std::cin >> n;

    while (n--) {
        std::cin >> x;

        if (is_prime(x)) {
            std::cout << x << " eh primo" << std::endl;
        } else {
            std::cout << x << " nao eh primo" << std::endl;
        }
    }

    return 0;
}

File: solutions/beecrowd/1165/1165.py
import math


def is_prime(n):
    if n == 2:
        return True
    elif n % 2 == 0 or n < 2:
        return False
    else:
        for divisor in range(3, int(math.sqrt(n)) + 1, 2):
            if n % divisor == 0:
                return False
    return True


n = int(input())

for _ in range(n):
    x = int(input())
    print(f'{x} eh primo' if is_prime(x) else f'{x} nao eh primo')
